docs/vector-library.md

```diff
@@ -0,0 +1,134 @@
+`vector.ceil(vec: vector): vector`
+
+Applies `math.ceil` component-wise for each vector. Equivalent of: `vector.create(math.ceil(vec.x), etc)`.
+
+`vector.abs(vec: vector): vector`
+
+Applies `math.abs` component-wise for each vector. Equivalent of: `vector.create(math.abs(vec.x), etc)`.
+
+`vector.sign(vec: vector): vector`
+
+Applies `math.sign` component-wise for each vector. Equivalent of: `vector.create(math.sign(vec.x), etc)`.
+
+`vector.clamp(vec: vector, min: vector, max: vector): vector`
+
+Applies `math.clamp` component-wise for each vector. Equivalent of: `vector.create(math.clamp(vec.x, min.x, max.x), etc)`.
+
+`vector.max(...: vector): vector`
+
+Applies `math.max` component-wise for each vector. Equivalent of: `vector.create(math.max((...).x), etc)`.
+
+`vector.min(...: vector): vector`
+
+Applies `math.min` component-wise for each vector. Equivalent of: `vector.create(math.min((...).x), etc)`.
+
+---
+
+`vector.zero`
+
+Vector where `x=0, y=0, z=0, w?=0`.
+
+`vector.one`
+
+Vector where `x=1, y=1, z=1, w?=1`.
+
+---
+
+### Arithmetic operations
+
+Primitive operators for vectors are already implemented, so this RFC doesn't concern vector arithmetic.
+
+### Component access
+
+Component access is already supported in the VM using the fields 'x', 'y', 'z', 'X', 'Y' and 'Z' (plus 'w' and 'W' in 4-wide mode).
+Note that writes to a single component are not supported as the vector value is immutable.
+This RFC doesn't define any changes to this behavior.
+
+
+### Compiler options
+
+Currently, there are 2 compiler options relating to vectors, `vectorLib` & `vectorCtor`. This poses an interesting problem: a builtin library would remove the _requirement_ for such compiler options, however these options still need to be supported. The intuitive solution to this is to leave both compiler options working and maintained, but provide the built-in vector library by default, allowing two vector libraries and two vector constructors.
+
+### Typechecking
+
+A new primitive `vector` type should be added to the type system.
+
+## Future Work
+
+A better vector constructor with new syntax, such as `<x, y, z>` or `(x, y, z)` or `|x, y, z|` or `vector(x, y, z)` or `[x, y, z]` could be implemented. This would make code that does stuff with vectors much less verbose.
+
+The buffer library could include functions to work with vectors, such as `buffer.writevector2`, `buffer.writevector3`, and in 4-wide mode, `buffer.writevector4`. The corresponding read functions would of course be included.
+
+## Drawbacks
+
+As per all additional globals, this creates a new global `vector`. This is a commonly used variable name, and many style guides will advise to avoid using `vector` as a variable name due to the global being added.
+
+Introducing a vector library means introducing more globals. Due to the prescence of vectors in performance-intensive code, it can be assumed that the built-in functions will have fastcalls. This means more fastcall slots will be used.
```

Comment:
```suggestion
Introducing a vector library means introducing more globals. Due to the presence of vectors in performance-intensive code, it can be assumed that the built-in functions will have fastcalls. This means more fastcall slots will be used.
```